rakelib/release.rake

```diff
@@ -134,27 +153,48 @@ namespace :release do
     JSON.parse(response.body)
   end
 
+  def github_upload(json, filename)
+    upload_url = json['upload_url'].gsub(/\{.*\}/, "?name=#{filename}")
+    zipfile = "#{BUILD_DIR}/#{filename}"
+    zipsize = File.size(zipfile)
+
+    Utils.print_header "Uploading ZIP (#{zipsize} bytes)"
+    post(upload_url, 'application/zip') do |req|
+      req.body_stream = File.open(zipfile, 'rb')
+      req.add_field('Content-Length', zipsize)
+      req.add_field('Content-Transfer-Encoding', 'binary')
+    end
+  end
+
   desc 'Upload the zipped binaries to a new GitHub release'
-  task :github => :zip do
+  task :github => :artifactbundle do
     v = Utils.podspec_version('SwiftGen')
 
+    artifact_names = [
+      "swiftgen-#{v}.zip",
+      "swiftgen.artifactbundle.zip"
+    ]    
+
+    # Pick out the relevant CHANGELOG information for this release
     changelog = `sed -n /'^## #{v}$'/,/'^## '/p CHANGELOG.md`.gsub(/^## .*$/, '').strip
+
+    # Append checksums for our generated artifacts
+    changelog << "\n\n### Checksums\n"
+    for artifact_name in artifact_names
+      changelog << "\n- #{artifact_name} `#{Digest::SHA256.file("#{BUILD_DIR}/#{artifact_name}")}`"
+    end
+
     Utils.print_header "Releasing version #{v} on GitHub"
     puts changelog
 
+    # Create the release
     json = post('https://api.github.com/repos/SwiftGen/SwiftGen/releases', 'application/json') do |req|
       req.body = { tag_name: v, name: v, body: changelog, draft: false, prerelease: false }.to_json
     end
-
-    upload_url = json['upload_url'].gsub(/\{.*\}/, "?name=swiftgen-#{v}.zip")
-    zipfile = "#{BUILD_DIR}/swiftgen-#{v}.zip"
-    zipsize = File.size(zipfile)
-
-    Utils.print_header "Uploading ZIP (#{zipsize} bytes)"
-    post(upload_url, 'application/zip') do |req|
-      req.body_stream = File.open(zipfile, 'rb')
-      req.add_field('Content-Length', zipsize)
-      req.add_field('Content-Transfer-Encoding', 'binary')
+   
+    # Upload our artifacts
+    for artifact_name in artifact_names
+      github_upload(json, artifact_name)
```

Comment:
Assuming you go with my suggestion above, that call site would then look like this:
```suggestion
      github_upload(upload_url: json['upload_url'], filename: artifact_name)
```